### ISSUE TITLE
decimal time

### ISSUE THREAD

**Issue by LucasLarson** [LucasLarson/dotfiles#758]
Leah Neukirchen serves [decimal time using Ruby](https://leahneukirchen.org/dotfiles/bin/fdate);[^1] this utility should be available here, too, but should use only POSIX-defined tools and options

[^1]: ([archive](https://web.archive.org/web/0id_/leahneukirchen.org/dotfiles/bin/fdate))

**Comment by LucasLarson** [comment]
This works even on System V
```sh
#! /bin/sh
# get current 24-hour time with seconds
LC_ALL='C' command date -- '+%H %M %S' |
  command awk -- '{
    # get the percentage of seconds that have passed so far today
    percentage = ($1 * 60 * 60 + $2 * 60 + $3) / (24 * 60 * 60)
    # print the content after the decimal point but
    # insert colons after the tenths (decimal hours) and thousandths (decimal minutes)
    printf "%d:%02d:%02d\n", substr(percentage, 3, 1), substr(percentage, 4, 2), substr(percentage, 6, 2)
}'
```

**Comment by LucasLarson** [comment]
fixed in #766 and [`fa1514e171`](https://github.com/LucasLarson/dotfiles/commit/fa1514e17138e2b0a0f0a0358ec51d0329a2eaff)
